File: backend/src/main/java/com/fitness/fitnessaicoach/repository/BodyMetricsRepository.java
package com.fitness.fitnessaicoach.repository;

import com.fitness.fitnessaicoach.domain.BodyMetrics;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface BodyMetricsRepository extends JpaRepository<BodyMetrics, UUID> {

    Optional<BodyMetrics> findTopByUserIdOrderByDateDescIdDesc(UUID userId);
}

File: android/fitness-ai-coach-app/app/src/main/java/com/fitness/fitnessaicoach/domain/repository/BodyMetricsRepository.kt
package com.fitness.fitnessaicoach.domain.repository

import com.fitness.fitnessaicoach.core.result.AppResult
import com.fitness.fitnessaicoach.domain.model.BodyMetrics

interface BodyMetricsRepository {
    suspend fun createBodyMetrics(bodyMetrics: BodyMetrics): AppResult<BodyMetrics>
    suspend fun getBodyMetrics(): AppResult<List<BodyMetrics>>
}
